auth/auth.go

```diff
@@ -442,14 +450,41 @@ func (auth *Authenticator) casUpdatePrincipal(p Principal, callback casUpdatePri
 }
 
 // Deletes a user/role.
-func (auth *Authenticator) Delete(p Principal) error {
-	if user, ok := p.(User); ok {
+func (auth *Authenticator) Delete(p Principal, purge bool, deleteSeq uint64) error {
+	user, ok := p.(User)
+	if ok {
 		if user.Email() != "" {
 			if err := auth.bucket.Delete(docIDForUserEmail(user.Email())); err != nil {
 				base.Debugf(base.KeyAuth, "Error deleting document ID for user email %s. Error: %v", base.UD(user.Email()), err)
 			}
 		}
+		return auth.bucket.Delete(p.DocID())
+	}
+
+	// If not a user, ie. a role
+	if purge {
+		return auth.Purge(p)
 	}
+	return auth.casUpdatePrincipal(p, func(p Principal) (updatedPrincipal Principal, err error) {
```

Comment:
casUpdatePrincipal doesn't look like it's using GetRoleIncDeleted - shouldn't it be doing that?  Do we have any tests for a role update that goes into CAS retry because the role has been deleted?

Reply:
If its deleted I think we still want to return nil here so shouldn't use GetRoleIncDeleted. I can try to write a test for this.

Reply:
Will write a test but pretty sure we want to get GetRole here so its nil in the case of a delete. This was the previous behaviour and then the callback would handle what to do in the situation where the value is nil.